### PR TITLE
nanobind v2.1.0

### PR DESCRIPTION
**Pull request by wjakob**
Adapted version of #16.

### CODE REVIEW

**Comment by wjakob** [comment]
@henryiii, could I ask for your help with this? Somehow the conda-forge build breaks after the nanobind transition to scikit-build-core. It does not use/recognize ninja.

**Comment by henryiii** [comment]
It is really frustrating, but conda-build hardcodes `CMAKE_GENERATOR` to `"Unix Makefiles"` even if make is not installed. You have two options: you can request make instead of ninja in your host table, or you can override CMAKE_GENERATOR with "Ninja" (or unset it). Most of the packages I know just request make.

**Comment by wjakob** [comment]
Thank you for clarifying! I don't know how to unset the variable and have chosen the `make` approach.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,19 +7,22 @@ package:
...
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
-  number: 1
+  number: 0
 
 requirements:
-  host:
+  build:
     - python >=3.8
+    - scikit-build-core
```
Comment (henryiii) [inline recipe/meta.yaml:23]: You need to list cmake here too. I can't make scikit-build-core depend on things like cmake and make due to issues with how conda works. :(

**Comment by henryiii** [comment]
By the way, you must work from a fork for conda-forge. Conda-forge deploys everything built from the repository itself - branches are assumed to be features or alternate versions.

**Pull request closed without merging**